smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/handlers/HttpBodyMiddleware.kt

```diff
@@ -107,7 +109,19 @@ class HttpBodyMiddleware(
                 writer.openBlock("do {", "} catch let err {") {
                     writer.write("let encoder = context.getEncoder()")
                     writer.openBlock("if let $memberName = input.operationInput.$memberName {", "} else {") {
-                        writer.write("let $dataDeclaration = try encoder.encode(\$L)", memberName)
+
+                        val xmlNameTrait = binding.member.getTrait<XmlNameTrait>() ?: target.getTrait<XmlNameTrait>()
+                        if (xmlNameTrait != null) {
+                            val xmlName = xmlNameTrait.value
+                            writer.write("let xmlEncoder = encoder as! XMLEncoder")
```

Comment:
Please explain why we should be confident force-casting here will never fail

Reply:
Good callout, will verify that the service protocol is xml here